services/ui-src/src/utils/autosave/helpers.ts

```diff
@@ -0,0 +1,50 @@
+import { FieldValues, UseFormReturn } from "react-hook-form";
+import { Choice, DropdownChoice, EntityShape, ReportStatus } from "types";
+
+const valueChanged = (currentValue: any, storedValue: any) => {
+  return currentValue !== storedValue;
+};
+
+export const shouldAutosave = (
+  currentValue: any,
+  storedValue: any,
+  autosave?: boolean,
+  isStateRep?: boolean,
+  isStateUser?: boolean
```

Comment:
Nit: These three args should be required, and `isStateuser` and `isStateRep` should be passed in as `isAuthorizedUser: isStateRep || isStateUser` from the component so there's one less.